pytorch_lightning/loggers/neptune.py

```diff
@@ -363,10 +368,14 @@ def _create_or_get_experiment(self):
             session = neptune.Session.with_default_backend(api_token=self.api_key)
             project = session.get_project(self.project_name)
 
-        if self._experiment_id is None:
+        if self.experiment_id is None:
             exp = project.create_experiment(name=self.experiment_name, **self._kwargs)
         else:
-            exp = project.get_experiments(id=self._experiment_id)[0]
+            exp = project.get_experiments(id=self.experiment_id)[0]
+            self.experiment_name = exp.get_system_properties()['name']
+            self.params = exp.get_parameters()
+            self.properties = exp.get_properties()
+            self.tags = exp.get_tags()
 
-        self._experiment_id = exp.id
+        self.experiment_id = exp.id
```

Comment:
why this assignment if the `self.experiment_id` is not empty?

Reply:
I suppose that's the only way ```self.experiment_id``` won't be empty inside ```_create_or_get_experiment()``` is passing it as an argument during a class initialization.  
But if it's passed as an argument ```exp = project.get_experiments(id=self.experiment_id)[0]``` will retrieve full information about the experiment with this experiment_id and ```exp.id``` will be equal to ```self.experiment_id``` as a result.  
Actually this assigment is needed only when new experiment was created. So, we can rearrange code like this:
```python
    def _create_or_get_experiment(self):
        if self.offline_mode:
            project = neptune.Session(backend=neptune.OfflineBackend()).get_project('dry-run/project')
        else:
            session = neptune.Session.with_default_backend(api_token=self.api_key)
            project = session.get_project(self.project_name)

        if self.experiment_id is None:
            exp = project.create_experiment(name=self.experiment_name, **self._kwargs)
            self.experiment_id = exp.id #<-------move assigment here 
        else:
            exp = project.get_experiments(id=self.experiment_id)[0]
            self.experiment_name = exp.get_system_properties()['name']
            self.params = exp.get_parameters()
            self.properties = exp.get_properties()
            self.tags = exp.get_tags()

        return exp```